### ISSUE TITLE
the error of training cogvideox-1.5-i2v

### ISSUE THREAD

**Issue by zzk88862** [a-r-r-o-w/finetrainers#117]
hi, great work!
now i have one machine for one A100 gpu with 80g,  and i have set num_processes=1， and export CUDA_VISIBLE_DEVICES=0，when i run train_image_to_video_sft.sh, the fllowing error happened, 

![image](https://github.com/user-attachments/assets/505a950a-7e8b-43e5-aafb-e3f4baf91439)

what should i do to solve this error?
tks

**Comment by jiashenggu** [comment]
Weird, only these info? Normally there should be an error before a distributed error.

**Comment by zzk88862** [comment]
yes, do y know how to solve it?

**Comment by gulucaptain** [comment]
same error, how to fix it?

**Comment by jiashenggu** [comment]
> yes, do y know how to solve it?
Could you provide your script?

**Comment by gulucaptain** [comment]
> same error, how to fix it?

I think the problem is due to the accelerate envs, because I use two gpus, but accelerate.device only contains cuda:0

**Comment by gulucaptain** [comment]
I wonder what's your CUDA version?

**Comment by zzk88862** [comment]
> > yes, do y know how to solve it?
> 
> Could you provide your script?

ok, this,
#!/bin/bash
export TORCH_LOGS="+dynamo,recompiles,graph_breaks"
export TORCHDYNAMO_VERBOSE=1
export WANDB_MODE="offline"
# export NCCL_P2P_DISABLE=1
export TORCH_NCCL_ENABLE_MONITORING=0
export TOKENIZERS_PARALLELISM=true
export OMP_NUM_THREADS=16
export CUDA_VISIBLE_DEVICES=0

# GPU_IDS="0"

# Training Configurations
# Experiment with as many hyperparameters as you want!
LEARNING_RATES=("1e-4")
LR_SCHEDULES=("cosine_with_restarts")
OPTIMIZERS=("adamw")
MAX_TRAIN_STEPS=("20000")

# Single GPU uncompiled training
ACCELERATE_CONFIG_FILE="accelerate_configs/deepspeed.yaml"

# Absolute path to where the data is located. Make sure to have read the README for how to prepare data.
# This example assumes you downloaded an already prepared dataset from HF CLI as follows:
#   huggingface-cli download --repo-type dataset Wild-Heart/Disney-VideoGeneration-Dataset --local-dir /path/to/my/datasets/disney-dataset
DATA_ROOT="./video-dataset-disney"
CAPTION_COLUMN="prompts.txt"
VIDEO_COLUMN="videos.txt"
MODEL_PATH="./THUDM/CogVideoX1.5-5B-I2V"

# Set ` --load_tensors ` to load tensors from disk instead of recomputing the encoder process.
# Launch experiments with different hyperparameters

for learning_rate in "${LEARNING_RATES[@]}"; do
  for lr_schedule in "${LR_SCHEDULES[@]}"; do
    for optimizer in "${OPTIMIZERS[@]}"; do
      for steps in "${MAX_TRAIN_STEPS[@]}"; do
        output_dir="./cogvideox-sft__optimizer_${optimizer}__steps_${steps}__lr-schedule_${lr_schedule}__learning-rate_${learning_rate}/"

        cmd="accelerate launch --config_file $ACCELERATE_CONFIG_FILE \
          training/cogvideox_image_to_video_sft.py \
          --pretrained_model_name_or_path  $MODEL_PATH \
          --data_root $DATA_ROOT \
          --caption_column $CAPTION_COLUMN \
          --video_column $VIDEO_COLUMN \
          --id_token BW_STYLE \
          --height_buckets 480 \
          --width_buckets 720 \
          --frame_buckets 49 \
          --dataloader_num_workers 8 \
          --pin_memory \
          --validation_prompt \"BW_STYLE A black and white animated scene unfolds with an anthropomorphic goat surrounded by musical notes and symbols, suggesting a playful environment. Mickey Mouse appears, leaning forward in curiosity as the goat remains still. The goat then engages with Mickey, who bends down to converse or react. The dynamics shift as Mickey grabs the goat, potentially in surprise or playfulness, amidst a minimalistic background. The scene captures the evolving relationship between the two characters in a whimsical, animated setting, emphasizing their interactions and emotions:::BW_STYLE A panda, dressed in a small, red jacket and a tiny hat, sits on a wooden stool in a serene bamboo forest. The panda's fluffy paws strum a miniature acoustic guitar, producing soft, melodic tunes. Nearby, a few other pandas gather, watching curiously and some clapping in rhythm. Sunlight filters through the tall bamboo, casting a gentle glow on the scene. The panda's face is expressive, showing concentration and joy as it plays. The background includes a small, flowing stream and vibrant green foliage, enhancing the peaceful and magical atmosphere of this unique musical performance\" \
          --validation_images \"/path/to/image1.png:::/path/to/image2.png\" \
          --validation_prompt_separator ::: \
          --num_validation_videos 1 \
          --validation_epochs 1 \
          --seed 42 \
          --mixed_precision bf16 \
          --output_dir $output_dir \
          --max_num_frames 77 \
          --train_batch_size 1 \
          --max_train_steps $steps \
          --checkpointing_steps 2000 \
          --gradient_accumulation_steps 4 \
          --gradient_checkpointing \
          --learning_rate $learning_rate \
          --lr_scheduler $lr_schedule \
          --lr_warmup_steps 800 \
          --lr_num_cycles 1 \
          --enable_slicing \
          --enable_tiling \
          --noised_image_dropout 0.05 \
          --optimizer $optimizer \
          --beta1 0.9 \
          --beta2 0.95 \
          --weight_decay 0.001 \
          --max_grad_norm 1.0 \
          --allow_tf32 \
          --report_to wandb \
          --nccl_timeout 1800"
        
        echo "Running command: $cmd"
        eval $cmd
        echo -ne "-------------------- Finished executing script --------------------\n\n"
      done
    done
  done
done

**Comment by jiashenggu** [comment]
> #!/bin/bash export TORCH_LOGS="+dynamo,recompiles,graph_breaks" export TORCHDYNAMO_VERBOSE=1 export WANDB_MODE="offline"
> 
> # export NCCL_P2P_DISABLE=1
> export TORCH_NCCL_ENABLE_MONITORING=0 export TOKENIZERS_PARALLELISM=true export OMP_NUM_THREADS=16 export CUDA_VISIBLE_DEVICES=0
> 
> # GPU_IDS="0"
> # Training Configurations
> # Experiment with as many hyperparameters as you want!
> LEARNING_RATES=("1e-4") LR_SCHEDULES=("cosine_with_restarts") OPTIMIZERS=("adamw") MAX_TRAIN_STEPS=("20000")
> 
> # Single GPU uncompiled training
> ACCELERATE_CONFIG_FILE="accelerate_configs/deepspeed.yaml"
> 
> # Absolute path to where the data is located. Make sure to have read the README for how to prepare data.
> # This example assumes you downloaded an already prepared dataset from HF CLI as follows:
> # huggingface-cli download --repo-type dataset Wild-Heart/Disney-VideoGeneration-Dataset --local-dir /path/to/my/datasets/disney-dataset
> DATA_ROOT="./video-dataset-disney" CAPTION_COLUMN="prompts.txt" VIDEO_COLUMN="videos.txt" MODEL_PATH="./THUDM/CogVideoX1.5-5B-I2V"
> 
> # Set `--load_tensors` to load tensors from disk instead of recomputing the encoder process.
> # Launch experiments with different hyperparameters
> for learning_rate in "${LEARNING_RATES[@]}"; do for lr_schedule in "${LR_SCHEDULES[@]}"; do for optimizer in "${OPTIMIZERS[@]}"; do for steps in "${MAX_TRAIN_STEPS[@]}"; do output_dir="./cogvideox-sft__optimizer_${optimizer}__steps_${steps}__lr-schedule_${lr_schedule}__learning-rate_${learning_rate}/"
> 
> ```
>     cmd="accelerate launch --config_file $ACCELERATE_CONFIG_FILE \
>       training/cogvideox_image_to_video_sft.py \
>       --pretrained_model_name_or_path  $MODEL_PATH \
>       --data_root $DATA_ROOT \
>       --caption_column $CAPTION_COLUMN \
>       --video_column $VIDEO_COLUMN \
>       --id_token BW_STYLE \
>       --height_buckets 480 \
>       --width_buckets 720 \
>       --frame_buckets 49 \
>       --dataloader_num_workers 8 \
>       --pin_memory \
>       --validation_prompt \"BW_STYLE A black and white animated scene unfolds with an anthropomorphic goat surrounded by musical notes and symbols, suggesting a playful environment. Mickey Mouse appears, leaning forward in curiosity as the goat remains still. The goat then engages with Mickey, who bends down to converse or react. The dynamics shift as Mickey grabs the goat, potentially in surprise or playfulness, amidst a minimalistic background. The scene captures the evolving relationship between the two characters in a whimsical, animated setting, emphasizing their interactions and emotions:::BW_STYLE A panda, dressed in a small, red jacket and a tiny hat, sits on a wooden stool in a serene bamboo forest. The panda's fluffy paws strum a miniature acoustic guitar, producing soft, melodic tunes. Nearby, a few other pandas gather, watching curiously and some clapping in rhythm. Sunlight filters through the tall bamboo, casting a gentle glow on the scene. The panda's face is expressive, showing concentration and joy as it plays. The background includes a small, flowing stream and vibrant green foliage, enhancing the peaceful and magical atmosphere of this unique musical performance\" \
>       --validation_images \"/path/to/image1.png:::/path/to/image2.png\" \
>       --validation_prompt_separator ::: \
>       --num_validation_videos 1 \
>       --validation_epochs 1 \
>       --seed 42 \
>       --mixed_precision bf16 \
>       --output_dir $output_dir \
>       --max_num_frames 77 \
>       --train_batch_size 1 \
>       --max_train_steps $steps \
>       --checkpointing_steps 2000 \
>       --gradient_accumulation_steps 4 \
>       --gradient_checkpointing \
>       --learning_rate $learning_rate \
>       --lr_scheduler $lr_schedule \
>       --lr_warmup_steps 800 \
>       --lr_num_cycles 1 \
>       --enable_slicing \
>       --enable_tiling \
>       --noised_image_dropout 0.05 \
>       --optimizer $optimizer \
>       --beta1 0.9 \
>       --beta2 0.95 \
>       --weight_decay 0.001 \
>       --max_grad_norm 1.0 \
>       --allow_tf32 \
>       --report_to wandb \
>       --nccl_timeout 1800"
>     
>     echo "Running command: $cmd"
>     eval $cmd
>     echo -ne "-------------------- Finished executing script --------------------\n\n"
>   done
> done
> ```
> 
> done done
You can try replacing deepspeed.yaml with uncompiled_1.yaml. It may make more sense to use uncompiled_1.yaml in a single-GPU environment.

**Comment by gulucaptain** [comment]
Thanks, I have solved this problem.

**Comment by zzk88862** [comment]
> Thanks, I have solved this problem.

hi，how did you solve this problem?

**Comment by gulucaptain** [comment]
> > Thanks, I have solved this problem.
> 
> hi，how did you solve this problem?

please check your cuda version and "swap memory" of linux, this problem is mainly caused by hardware.

**Comment by sayakpaul** [comment]
Closing due to inactivity. Please feel free to reopen.
